lib/spack/docs/module_file_support.rst

```diff
@@ -0,0 +1,664 @@
+   for more details.
+
+----------------------
+Activate shell support
+----------------------
+
+You can enable shell support by sourcing the appropriate setup file
+in the ``$SPACK_ROOT/share/spack`` directory.
+For ``bash`` or ``ksh`` users:
+
+.. code-block:: console
+
+   $ . ${SPACK_ROOT}/share/spack/setup-env.sh
+
+For ``csh`` and ``tcsh`` instead:
+
+.. code-block:: console
+
+   $ source $SPACK_ROOT/share/spack/setup-env.csh
+
+
+.. note::
+  You can put the source line in your ``.bashrc`` or ``.cshrc`` to
+  have Spack's shell support available on the command line at any login.
+
+
+----------------------------
+Using module files via Spack
+----------------------------
+
+If you have shell support enabled you should be able to run either
+``module avail`` or ``use -l spack`` to see what module/dotkit files have
+been installed.  Here is sample output of those programs, showing lots
+of installed packages.
+
+.. code-block:: console
+
+   $ module avail
+
+   ------- /home/gamblin2/spack/share/spack/modules/linux-debian7-x86_64 --------
+   adept-utils@1.0%gcc@4.4.7-5adef8da   libelf@0.8.13%gcc@4.4.7
+   automaded@1.0%gcc@4.4.7-d9691bb0     libelf@0.8.13%intel@15.0.0
+   boost@1.55.0%gcc@4.4.7               mpc@1.0.2%gcc@4.4.7-559607f5
+   callpath@1.0.1%gcc@4.4.7-5dce4318    mpfr@3.1.2%gcc@4.4.7
+   dyninst@8.1.2%gcc@4.4.7-b040c20e     mpich@3.0.4%gcc@4.4.7
+   gcc@4.9.1%gcc@4.4.7-93ab98c5         mpich@3.0.4%gcc@4.9.0
+   gmp@6.0.0a%gcc@4.4.7                 mrnet@4.1.0%gcc@4.4.7-72b7881d
+   graphlib@2.0.0%gcc@4.4.7             netgauge@2.4.6%gcc@4.9.0-27912b7b
+   launchmon@1.0.1%gcc@4.4.7            stat@2.1.0%gcc@4.4.7-51101207
+   libNBC@1.1.1%gcc@4.9.0-27912b7b      sundials@2.5.0%gcc@4.9.0-27912b7b
+   libdwarf@20130729%gcc@4.4.7-b52fac98
+
+.. code-block:: console
+
+   $ use -l spack
+
+   spack ----------
+     adept-utils@1.0%gcc@4.4.7-5adef8da - adept-utils @1.0
+     automaded@1.0%gcc@4.4.7-d9691bb0 - automaded @1.0
+     boost@1.55.0%gcc@4.4.7 - boost @1.55.0
+     callpath@1.0.1%gcc@4.4.7-5dce4318 - callpath @1.0.1
+     dyninst@8.1.2%gcc@4.4.7-b040c20e - dyninst @8.1.2
+     gmp@6.0.0a%gcc@4.4.7 - gmp @6.0.0a
+     libNBC@1.1.1%gcc@4.9.0-27912b7b - libNBC @1.1.1
+     libdwarf@20130729%gcc@4.4.7-b52fac98 - libdwarf @20130729
+     libelf@0.8.13%gcc@4.4.7 - libelf @0.8.13
+     libelf@0.8.13%intel@15.0.0 - libelf @0.8.13
+     mpc@1.0.2%gcc@4.4.7-559607f5 - mpc @1.0.2
+     mpfr@3.1.2%gcc@4.4.7 - mpfr @3.1.2
+     mpich@3.0.4%gcc@4.4.7 - mpich @3.0.4
+     mpich@3.0.4%gcc@4.9.0 - mpich @3.0.4
+     netgauge@2.4.6%gcc@4.9.0-27912b7b - netgauge @2.4.6
+     sundials@2.5.0%gcc@4.9.0-27912b7b - sundials @2.5.0
+
+The names here should look familiar, they're the same ones from
+``spack find``.  You *can* use the names here directly.  For example,
+you could type either of these commands to load the callpath module:
+
+.. code-block:: console
+
+   $ use callpath@1.0.1%gcc@4.4.7-5dce4318
+
```

Comment:
Are these hashes explained in the `spack find` docs?  This is a new feature for me...